paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentOptionsActivityTest.kt

```diff
@@ -355,53 +331,22 @@ internal class PaymentOptionsActivityTest {
         assertThat(scenario.state).isEqualTo(Lifecycle.State.DESTROYED)
     }
 
-    @Test
-    fun `Clears error on user selection`() {
-        val scenario = activityScenario()
-        scenario.launch(createIntent()).onActivity { activity ->
-            val text = "some error"
-            viewModel.onError(text)
-
-            composeTestRule
-                .onNodeWithText(text)
-                .assertIsDisplayed()
-
-            viewModel.onUserSelection()
-
-            composeTestRule
-                .onNodeWithText(text)
-                .assertDoesNotExist()
-        }
-    }
-
-    private fun createIntent(
-        args: PaymentOptionContract.Args = PAYMENT_OPTIONS_CONTRACT_ARGS
-    ): Intent {
-        return Intent(
+    private fun runActivityScenario(
+        args: PaymentOptionContract.Args = PAYMENT_OPTIONS_CONTRACT_ARGS,
+        block: (InjectableActivityScenario<PaymentOptionsActivity>) -> Unit,
+    ) {
+        val intent = Intent(
             ApplicationProvider.getApplicationContext(),
             PaymentOptionsActivity::class.java
         ).putExtras(
             bundleOf(ActivityStarter.Args.EXTRA to args)
         )
-    }
-
-    private fun activityScenario(
-        viewModel: PaymentOptionsViewModel = this.viewModel
-    ): InjectableActivityScenario<PaymentOptionsActivity> {
-        return injectableActivityScenario {
-            injectActivity {
-                viewModelFactory = viewModelFactoryFor(viewModel)
-            }
-        }
-    }
 
-    private fun createViewModel(
-        args: PaymentOptionContract.Args = PAYMENT_OPTIONS_CONTRACT_ARGS
-    ): PaymentOptionsViewModel {
```

Comment:
I deliberately moved the contents of this method into `runActivityScenario` to discourage direct usage.